File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/agent/AgentPrepAnonymousMsgTest.java
package org.hyperledger.indy.sdk.agent;

import org.hyperledger.indy.sdk.IndyIntegrationTestWithSingleWallet;
import org.hyperledger.indy.sdk.InvalidStructureException;
import org.hyperledger.indy.sdk.signus.Signus;
import org.hyperledger.indy.sdk.signus.SignusJSONParameters;
import org.hyperledger.indy.sdk.signus.SignusResults;
import org.hyperledger.indy.sdk.wallet.Wallet;
import org.json.JSONObject;
import org.junit.Test;

import java.util.Arrays;
import java.util.Base64;
import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.*;

public class AgentPrepAnonymousMsgTest extends IndyIntegrationTestWithSingleWallet {

	private void checkMessage(byte[] encryptedMsg) throws Exception {
		Wallet.createWallet(POOL, "walletForCheck", TYPE, null, null).get();
		Wallet localWallet = Wallet.openWallet("walletForCheck", null, null).get();

		String didJson = new SignusJSONParameters.CreateAndStoreMyDidJSONParameter(null, MY1_SEED, null, false).toJson();
		SignusResults.CreateAndStoreMyDidResult result = Signus.createAndStoreMyDid(localWallet, didJson).get();
		String recipientDid = result.getDid();

		byte[] decryptedMessage = Signus.decryptSealed(localWallet, recipientDid, encryptedMsg).get();
		String decryptedMessageJson = new String(decryptedMessage);
		JSONObject decryptedMsg = new JSONObject(decryptedMessageJson);

		assertFalse(decryptedMsg.getBoolean("auth"));
		assertTrue(Arrays.equals(MESSAGE, Base64.getDecoder().decode(decryptedMsg.getString("msg"))));

		localWallet.closeWallet().get();
		Wallet.deleteWallet("walletForCheck", null).get();
	}

	@Test
	public void testPrepAnonymousMsgWorks() throws Exception {
		byte[] encryptedMsg = Agent.prepAnonymousMsg(VERKEY_MY1, MESSAGE).get();
		checkMessage(encryptedMsg);
	}

	@Test
	public void testPrepAnonymousMsgWorksForInvalidRecipientVk() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(InvalidStructureException.class));

		Agent.prepAnonymousMsg(INVALID_VERKEY, MESSAGE).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/AgentTests/AgentPrepAnonymousMsgTest.cs
using Hyperledger.Indy.AgentApi;
using Hyperledger.Indy.SignusApi;
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Newtonsoft.Json.Linq;
using System;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.AgentTests
{
    [TestClass]
    public class AgentPrepAnonymousMsgTest : IndyIntegrationTestWithSingleWallet
    {
        private async Task CheckMessage(byte[] encryptedMsg)
        {
            await Wallet.CreateWalletAsync(POOL, "walletForCheck", TYPE, null, null);
            var localWallet = await Wallet.OpenWalletAsync("walletForCheck", null, null);

            var didJson = string.Format("{{\"seed\":\"{0}\",\"cid\":false}}", MY1_SEED);
            var result = await Signus.CreateAndStoreMyDidAsync(localWallet, didJson);
            var recipientDid = result.Did;

            var decryptedMessageBytes = await Signus.DecryptSealedAsync(localWallet, recipientDid, encryptedMsg);
            var decryptedMessageJson = Encoding.UTF8.GetString(decryptedMessageBytes);
            var decryptedMsg = JObject.Parse(decryptedMessageJson);


            Assert.IsFalse(decryptedMsg.Value<bool>("auth"));
            Assert.IsTrue(MESSAGE.SequenceEqual(Convert.FromBase64String(decryptedMsg.Value<string>("msg"))));

            await localWallet.CloseAsync();
            await Wallet.DeleteWalletAsync("walletForCheck", null);            
        }

        [TestMethod]
        public async Task TestPrepAnonymousMsgWorks()
        {
            var encryptedMsg = await Agent.PrepAnonymousMsgAsync(VERKEY_FOR_MY1_SEED, MESSAGE);
            await CheckMessage(encryptedMsg);
        }

        [TestMethod]
        public async Task TestPrepAnonymousMsgWorksForInvalidRecipientVk()
        {
            var ex = await Assert.ThrowsExceptionAsync<InvalidStructureException>(() =>
                Agent.PrepAnonymousMsgAsync(INVALID_VERKEY, MESSAGE)
            );
        }
    }
}
